beluga_ros/tools/conversion_utils.py

```diff
@@ -0,0 +1,273 @@
+
+            for idx, distribution in enumerate(self._grid.values()):
+                means_dataset[idx] = distribution.mean
+                covariances_dataset[idx] = distribution.covariance
+            fp.create_dataset("resolution", data=np.asarray(self._resolution))
+
+    @classmethod
+    def from_hdf5(cls, hdf5_file: Path):
+        """
+        Create an NDTMap instance from a path to a hdf5 file.
+
+        See 'to_hdf5' docstring for details on the hdf5 format.
+        """
+        with h5py.File(hdf5_file.absolute(), "r") as fp:
+            resolution: float = fp["resolution"][()]
+            means: np.ndarray = fp["means"]
+            discrete_cells: np.ndarray = fp["discrete_cells"]
+            covs: np.ndarray = fp["covariances"]
+            total_cells = covs.shape[0]
+
+            assert covs.shape[1] == 2
+            assert covs.shape[2] == 2
+            assert means.shape[1] == 2
+            assert discrete_cells.shape[0] == total_cells
+            assert discrete_cells.shape[1] == 2
+            assert means.shape[0] == total_cells
+
+            ret = NDTMap(resolution=resolution)
+            for cell_num in range(total_cells):
+                cell = DiscreteCell(*discrete_cells[cell_num].tolist())
+                mean = means[cell_num]
+                cov = covs[cell_num]
+                ret.add_distribution(
+                    cell=cell, ndt=NormalDistribution(mean=mean, covariance=cov)
+                )
+            return ret
+
+
+@dataclass
+class OccupancyGrid:
+    resolution: float
+    origin: np.ndarray
+    grid: np.ndarray
+
+    @staticmethod
+    def load_from_file(yaml_path: Path) -> "OccupancyGrid":
+        """Create an grid from a yaml file describing a ROS style occupancy grid."""
+        assert yaml_path.is_file(), f"file {yaml_path} doesn't exist."
+        assert yaml_path.suffix == ".yaml"
+
+        with open(yaml_path, "rb") as fp:
+            data = yaml.safe_load(fp)
+
+        pgm_path: Path = yaml_path.parent / data["image"]
+        assert pgm_path.is_file(), f".pgm file at {pgm_path} does not exist"
+
+        with open(yaml_path.parent / data["image"], "rb") as pgmf:
+            grid = plt.imread(pgmf)
+
+        return OccupancyGrid(
+            resolution=data["resolution"], origin=np.asarray(data["origin"]), grid=grid
+        )
+
+
+def grid_to_point_cloud(occupancy_grid: OccupancyGrid) -> np.ndarray:
+    """
+    Convert an OccupancyGrid's occupied cells to a 2D point cloud.
+
+    Uses the center of the cell for the conversion to reduce max error.
+    """
+    occupied_cells_indices = np.asarray(
+        np.where(occupancy_grid.grid == 0)
+    )  # in ROS maps 0 == occupied.
+    res = occupancy_grid.resolution
+
+    # Discretized occupied cells using the center of the cell.
+    points = occupied_cells_indices * res + np.ones_like(occupied_cells_indices) * (
+        res / 2
+    )
```

Comment:
@serraramiro1 nit, fyi: `numpy` broadcasts when summing a `numpy` array and a scalar:
```suggestion
    points = occupied_cells_indices * res + res / 2
```

Reply:
That's a new one for me. Done!